app/queues/notifications/expiration-notification.server.ts

```diff
@@ -0,0 +1,189 @@
+import type { NotificationData } from './notifications.server';
+
+enum Type {
+  Certificate = 'certificate',
+  Record = 'record',
+}
+
+// name for the queue
+const expirationNotificationQueueName = 'expiration-notification';
+
+const updateNotificationStatus = (type: string, id: number) => {
+  // throw if any of the arguments are missing
+  if (!type || !id) {
+    throw new Error(
+      'arguments: type and id for updateNotificationStatus cannot be null or undefined'
+    );
+  }
+  switch (type) {
+    case Type.Certificate:
+      return prisma.certificate.update({
+        where: {
+          id,
+        },
+        data: {
+          lastNotified: new Date(),
+        },
+      });
+    case Type.Record:
+      return prisma.record.update({
+        where: {
+          id,
+        },
+        data: {
+          lastNotified: new Date(),
+        },
+      });
+  }
+};
+
+// fetch records with by status
+const getRecordsByExpiration = (type: string) => {
+  let monthFromToday = new Date();
+  monthFromToday.setMonth(monthFromToday.getMonth() + 1);
+  if (!type) {
+    throw new Error('argument: type for getRecordsByExpiration cannot be null or undefined');
+  }
+  switch (type) {
+    case Type.Certificate:
+      return prisma.certificate.findMany({
+        where: {
+          AND: [
+            {
+              validTo: {
+                lte: monthFromToday,
+              },
+              lastNotified: null,
+            },
+          ],
+        },
+        select: {
+          user: true,
+          id: true,
+          lastNotified: true,
+        },
+      });
+    case Type.Record:
+      return prisma.record.findMany({
+        where: {
+          AND: [
+            {
+              expiresAt: {
+                lte: monthFromToday,
+              },
+              lastNotified: null,
+            },
+          ],
+        },
+        select: {
+          user: true,
+          id: true,
+          lastNotified: true,
+        },
+      });
+  }
+};
+// Queue initialization
+export const expirationNotificationQueue = new Queue<ExpirationStatusPayload[]>(
+  expirationNotificationQueueName,
+  {
+    connection: redis,
+  }
+);
+
+expirationNotificationQueue.on('error', (err) => {
+  logger.warn('Error running check for DNS Records/Certificates about to expire', err);
+});
+
+// function to add jobs
+export const addExpirationNotification = async (type: string) => {
+  let jobName: string;
+  let data: ExpirationStatusPayload[];
+
+  if (!type) {
+    throw new Error('argument: type for addExpirationCheckJobs cannot be null or undefined');
+  }
+  switch (type) {
+    case Type.Certificate:
+      data = await getRecordsByExpiration(Type.Certificate)!;
+      if (data.length) {
+        logger.info('Entering if');
+        jobName = `${expirationNotificationQueueName}-${
+          Type.Certificate
+        }:${new Date().toLocaleString('en-CA')}`;
```

Comment:
We have to be careful here.  The job name needs to be unique, but, it also needs to be the same per job.  The reason for this is that 5 mins later, we don't want to add yet another job for the same task.  Also, if we have multiple instances of the server running, we don't want to have both servers start identical jobs.  Make the job name unique by using the user's name, certificate id, etc. but not a date.